### ISSUE TITLE
DirectProduct incorrectly adding identity

### ISSUE THREAD

**Issue by mtorpey** [gap-packages/Semigroups#342]
`DirectProduct` is not behaving correctly.  It appears to be appending the identity transformation when it shouldn't.  Consider the following example (found by @ffloresbrito):

```gap
gap> S := Semigroup([Transformation([1, 2, 3, 3, 3])]);;
gap> CC := Semigroup([Transformation([2, 1])]);;
gap> Size(S); Size(CC);
1
2
gap> DP := DirectProduct(S, CC);;
gap> Elements(DP);
[ Transformation( [ 1, 2, 3, 3, 3 ] ), 
  Transformation( [ 1, 2, 3, 3, 3, 7, 6 ] ), IdentityTransformation, 
  Transformation( [ 1, 2, 3, 4, 5, 7, 6 ] ) ]
gap> Size(DP);
2
```

As you can see, the function seems to be treating `S` as though it contains `IdentityTransformation`, which it doesn't.  Hence two extra elements are ending up in the direct product, even though the size is being calculated correctly.

While I was at it, I noticed that `DirectProduct` for semigroups is not documented, meaning that the only manual entry for it is in the GAP manual, where it just talks about groups.  This should be rectified as well (I'll add it to Issue #298).

**Comment by james-d-mitchell** [comment]
Hmm, something is for sure not right here...

**Comment by ChrisJefferson** [comment]
What was the error?

**Comment by james-d-mitchell** [comment]
Don't know but a semigroup of size 2 shouldn't have 4 elements

**Comment by james-d-mitchell** [comment]
I've fixed this in PR #343

**Comment by mtorpey** [comment]
Merged and fixed!
